dev/com.ibm.ws.jpa_spec10_query_fat/test-applications/olgh8014/src/com/ibm/ws/jpa/olgh8014/testlogic/JPATestOLGH8014Logic.java

```diff
@@ -387,7 +387,7 @@ public void testAggregateFunctionsWithPrimitives(TestExecutionContext testExecCt
                 Assert.assertEquals(new Integer(20), res);
             } else {
                 //The specification defined assertion
-                Assert.assertEquals(new Double(20), res);
+                Assert.assertEquals(new Double(20).doubleValue(), (double) res, 0.5);
```

Comment:
You might have noticed the last argument, which allows for wriggle room.  Remember how float and decimal values are not precise, and can vary?  (hint: no reputable bank stores an account balance using decimals and floats)

Reply:
So you added the delta as a precaution, but there were no test failures that lead to this change?

Reply:
Without them, Oracle and SQLServer failed because the database did not store the float value exactly the same as as the JVM's value.  So when the query executed, the float value returned was difference.  Because 2.1 != 2.11.  So the third argument is a tolerance for that variance.

Reply:
Hmm. That seems like an issue. Customers aren't going to be ok with an arbitrary tolerance tho. Do you have the test results you could send me? Id like to take a look.

Reply:
Will, this is at the jdbc/database level.  Go ahead and try it with some raw JDBC.

Reply:
Maybe these tests are pretty useless then. They were meant to test precise return values, but every database returns a different result.